src/test/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageProcessorTest.java

```diff
@@ -0,0 +1,263 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.notificationservice.exception.InvalidMessageException;
+import uk.gov.hmcts.reform.notificationservice.model.request.incomming.NotificationMsg;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@ExtendWith(MockitoExtension.class)
+class NotificationMessageProcessorTest {
+
+
+    private NotificationMessageProcessor notificationMessageProcessor;
+
+    @Mock
+    private IMessageReceiver messageReceiver;
+    @Mock
+    private NotificationMessageHandler notificationMessageHandler;
+    @Mock
+    private NotificationMessageParser notificationMessageParser;
+
+    @BeforeEach
+    public void before() throws Exception {
+        notificationMessageProcessor = new NotificationMessageProcessor(
+            messageReceiver,
+            notificationMessageHandler,
+            notificationMessageParser,
+            5
+        );
+    }
+
+    @Test
+    public void should_return_true_when_there_is_a_message_to_process() throws Exception {
+        // given
+        IMessage message = mock(IMessage.class);
+        MessageBody messageBody = mock(MessageBody.class);
+        given(message.getMessageBody()).willReturn(messageBody);
+        given(messageReceiver.receive()).willReturn(message);
+
+        NotificationMsg notificationMsg = mock(NotificationMsg.class);
+        given(notificationMessageParser.parse(messageBody)).willReturn(notificationMsg);
+        doNothing().when(notificationMessageHandler).handleNotificationMessage(notificationMsg);
+
+        // when
+        boolean processedMessage = notificationMessageProcessor.processNextMessage();
+
+
+        // then
+        assertThat(processedMessage).isTrue();
+        verify(messageReceiver).receive();
+        verify(notificationMessageParser).parse(messageBody);
+        verify(notificationMessageHandler).handleNotificationMessage(notificationMsg);
+
+    }
+
+    @Test
+    public void should_return_false_when_there_is_no_message_to_process() throws Exception {
+        // given
+        given(messageReceiver.receive()).willReturn(null);
+
+        // when
+        boolean processedMessage = notificationMessageProcessor.processNextMessage();
+
+        // then
+        assertThat(processedMessage).isFalse();
```

Comment:
Could optionally check if there have been no interaction with the parser and the handler.

Reply:
added